src/__tests__/words.spec.ts

```diff
@@ -0,0 +1,297 @@
+import {describe, it, expect} from 'vitest';
+
+import {
+  add,
+  sub,
+  neg,
+  norm,
+  scalarMult,
+  rotate,
+  getStepVector,
+  stepSignature,
+  wordArity,
+  stepVectorArity,
+} from '../words';
+
+describe('add()', () => {
+  it('has identity element []', () => {
+    const vector = [
+      ['a', 1],
+      ['b', 2],
+      ['c', -3],
+    ];
+    const vectorClone = JSON.parse(JSON.stringify(vector));
```

Comment:
This trick is used all over the place here. Just make a `function clone(x) {return JSON.parse(JSON.stringify(x)}` helper above.